### PR TITLE
Fix clean_inactive for Filestream input

### PR DESCRIPTION
**Pull request by belimawr**
## Proposed commit message

The `clean_inactive` parameter was being parsed with the wrong key. It
is parsed/used by an anonymous struct on `input-logfile/manager.go`,
there it was parsed and used as `CleanTimeout` (`clean_timeout`). This
`CleanTimeout` setting has got exactly the same effect as the
`clean_inactive` described in our documentation.

This commit fixes this bug by renaming `clean_timeout` to
`clean_inactive` so the configuration value can have effect.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- ~~[ ] I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [x] Wait for https://github.com/elastic/beats/pull/38488 to be merged and rebase onto `main`

## How to test this PR locally
All files are at the end of this section.
1. Start Filebeat using `filebeat.yml`
2. Create the file to be ingested/write some data
    ```
    echo "foo bar" > /tmp/log.log
    ```
3. Look at Filebeat logs to ensure the file has been found, ingested and the reader was closed. Look for logs messages like those:
    ```
    A new file /tmp/log.log has been found
    Reader was closed. Closing. Path='/tmp/log.log
    ```
4. Ensure the store clean up run and removed the only entry, look for a log message like:
    ```
    1 entries removed
    ```
5. Ensure the registry log file (`data/registry/filebeat/log.json`) contains an remove operation like this:
    ```
    {"op":"remove","id":5}
    ```

<details><summary>filebeat.yml</summary>
<p>

```yaml
filebeat.inputs:
  - type: filestream
    id: "test-clean-inactive"
    paths:
      - /tmp/log.log

    clean_inactive: 3s
    ignore_older: 2s
    close.on_state_change.inactive: 1s
    prospector.scanner.check_interval: 1s

filebeat.registry:
  cleanup_interval: 5s
  flush: 1s

queue.mem:
  events: 32
  flush.min_events: 8
  flush.timeout: 0.1s

path.home: %s

output.file:
  path: ${path.home}
  filename: "output-file"
  rotate_every_kb: 10000

logging:
  level: debug
  selectors:
    - input
    - input.filestream
  metrics:
    enabled: false
```

</p>
</details> 

## Related issues
- Closes: https://github.com/elastic/beats/issues/36629
- Needs: https://github.com/elastic/beats/pull/38488

~~## Use cases~~
~~## Screenshots~~
~~## Logs~~

### CODE REVIEW

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -93,6 +93,9 @@ https://github.com/elastic/beats/compare/v8.8.1\...main[Check the HEAD diff]
 - Fix PEM key validation for CEL and HTTPJSON inputs. {pull}38405[38405]
 - Fix filebeat gcs input panic {pull}38407[38407]
 - Rename `activity_guid` to `activity_id` in ETW input events to suit other Windows inputs. {pull}38530[38530]
+- Fix `clean_removed` on Filestream input. {issue}36761[36761] {pull}38488[38488]
+- Fix filestream's registry GC: registry entries are now removed from the in-memory and disk store when they're older than the set TTL {issue}36761[36761] {pull}38488[38488]
```
Comment (rdner) [inline CHANGELOG.next.asciidoc:97]: These entries should not be here.

File: filebeat/input/filestream/internal/input-logfile/manager.go [hunk 2]
```diff
@@ -165,7 +165,7 @@ func (cim *InputManager) Create(config *conf.C) (v2.Input, error) {
 
 	settings := struct {
 		ID             string        `config:"id"`
-		CleanTimeout   time.Duration `config:"clean_timeout"`
+		CleanTimeout   time.Duration `config:"clean_inactive"`
```
Comment (rdner) [inline filebeat/input/filestream/internal/input-logfile/manager.go:168]: Let's also rename `CleanTimeout` -> `CleanInactive`, so we have less confusion mapping this field to the config parameter.

File: filebeat/input/v2/input-cursor/manager.go [hunk 3]
```diff
@@ -154,7 +154,7 @@ func (cim *InputManager) Create(config *conf.C) (v2.Input, error) {
 
 	settings := struct {
 		ID           string        `config:"id"`
-		CleanTimeout time.Duration `config:"clean_timeout"`
+		CleanTimeout time.Duration `config:"clean_inactive"`
```
Comment (rdner) [inline filebeat/input/v2/input-cursor/manager.go:157]: Same here, we should rename `CleanTimeout` -> `CleanInactive`

File: filebeat/tests/integration/filestream_test.go [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+	filebeat.WaitForLogs("filebeat start running", 20*time.Second, "did not find Filebeat start logs, did Filebeat start correctly?")
+
+	filebeat.WaitForLogs("Connection to backoff(elasticsearch(http://localhost:9200)", 2*time.Second, "did not connect to ES")
+	// filebeat.WaitForLogs("events have been published to elasticsearch in", 5*time.Second, "did not publish events to ES")
+
+	filebeat.WaitForLogs("removed state for", 30*time.Second, "did not find log entry about removing state from registry")
+
+
+*/
```
Comment (rdner) [inline filebeat/tests/integration/filestream_test.go:147]: Why is this code commented out?

**Review by rdner — CHANGES_REQUESTED** [review]
Also, the linter issue needs to be fixed.

File: filebeat/input/v2/input-cursor/manager.go (hunk 3 above)
Comment (belimawr) [reply filebeat/input/v2/input-cursor/manager.go]: Fixed by 2f7d96a02a

File: filebeat/tests/integration/filestream_test.go (hunk 4 above)
Comment (belimawr) [reply filebeat/tests/integration/filestream_test.go]: I forgot it there, thanks for catching this. I removed it: 2f7d96a02a

File: filebeat/docs/fields.asciidoc [hunk 5]
```diff
@@ -111136,6 +111136,13 @@ type: keyword
 
 --
 
+*`o365.audit.Activity`*::
```
Comment (belimawr) [inline filebeat/docs/fields.asciidoc:111139]: This is unrelated to the PR, but when I run `mage check` it got generated.

**Review by rdner — APPROVED** [review]

**Comment by nimarezainia** [comment]
Thank you all.

**Pull request merged**